auth/src/main/kotlin/com/hedvig/android/auth/NetworkAuthRepository.kt

```diff
@@ -33,19 +42,42 @@ class NetworkAuthRepository(
       .url("$url/member-login")
       .build()
 
-    return withContext(Dispatchers.IO) {
-      try {
-        okhttpClient.newCall(request)
-          .await()
-          .responseToResult()
-      } catch (e: java.lang.Exception) {
-        AuthAttemptResult.Error("Error: ${e.message}")
-      }
+    return try {
+      okhttpClient.newCall(request)
+        .await()
+        .toAuthAttemptResult()
+    } catch (e: java.lang.Exception) {
+      AuthAttemptResult.Error("Error: ${e.message}")
     }
   }
 
-  override fun observeLoginStatus(): Flow<LoginStatusResult> {
-    TODO("Not yet implemented")
+  override fun observeLoginStatus(statusUrl: String): Flow<LoginStatusResult> {
+    val request = Request.Builder()
+      .get()
+      .url("$url$statusUrl")
+      .build()
+
+    var isPending = false
+
+    return flow {
+      while (isPending) {
+        try {
+          val loginStatusResult = okhttpClient
+            .newCall(request)
+            .await()
+            .toLoginStatusResult()
+
+          emit(loginStatusResult)
+
+          isPending = loginStatusResult.isPending()
+          if (isPending) {
+            delay(POLL_DELAY_MILLIS)
+          }
```

Comment:
Hmm I *think* we can instead have this be a `while(true)` and when isPending is false we  `break` the look instead of relying on this mutable local state? I *think* there's no concurrency issues here, but it'd still be a bit more clear that way.